src/tests/conformance/query/basic.rs

```diff
@@ -0,0 +1,55 @@
+//! Contains test with basic queries.
+//!
+//! Queries and expected replies:
+//!
+//!     - Ping           -> Pong
+
+use crate::{
+    protocol::{
+        codecs::binary::{BinaryMessage, Payload},
+        proto::TmPing,
+    },
+    setup::node::Node,
+    tools::synth_node::SyntheticNode,
+};
+
+#[tokio::test]
+async fn ping() {
```

Comment:
For each test (or collections of cases depending on what makes sense), it would be good to add an entry to `SPEC.md` and referencing it here.
```suggestion
async fn ping() {
// ZG-CONFORMANCE-<insert case number here>
```